python/cudaq/kernel/ast_bridge.py

```diff
@@ -3330,6 +3361,54 @@ def visit_BinOp(self, node):
         else:
             self.emitFatalError(f"unhandled binary operator - {node.op}", node)
 
+    def __store_state(self, value: State):
+        # Compute a unique hash string for the state data
+        hashValue = hashlib.sha1(value).hexdigest(
```

Comment:
This reads from the state's buffer which causes a copy of the state from the gpu. Maybe just use a counter as an id?

Reply:
replaced by an id (just a counter)